dask/array/core.py

```diff
@@ -1074,7 +1075,7 @@ def store(
 
     if isinstance(sources, Array):
         sources = [sources]
-        targets = [targets]
+        targets = [targets]  # type: ignore
```

Comment:
This is wrong. I think the original author meant to say that you either give a list of sources and a list of targets, or a single source and a single target, but they didn't think about mix and matching. The code however is blindly assuming the two inputs are aligned and that's why mypy is - rightfully! - complaining.
```suggestion
    if isinstance(targets, Array):
        targets = [targets]
```